tools/pipelines/templates/build-docker-service.yml

```diff
@@ -575,10 +549,26 @@ extends:
           buildToolsVersionToInstall: ${{ parameters.buildToolsVersionToInstall }}
 
     - ${{ if eq(parameters.shouldReleaseDockerImage, true) }}:
-      - template: /tools/pipelines/templates/include-publish-docker-service.yml@self
-        parameters:
-          containerRegistry: $(containerRegistryConnection)
-          containerTag: $(containerTag)
-          containerRegistryUrl: $(containerRegistryUrl)
-          containerName: ${{ parameters.containerName }}
-          tagName: $(tagNameOnReleaseImage)
+      - stage: publish_docker_official
+        dependsOn: build
+        displayName: Publish Official Docker Image
+        condition: and(succeeded(), eq(variables['release'], 'release'))
+        variables:
+        - template: /tools/pipelines/templates/include-vars-docker.yml@self
+          parameters:
+            containerName: ${{ parameters.containerName }}
+            shouldPushDockerImage: ${{ parameters.shouldPushDockerImage }}
+            testBuild: ${{ parameters.testBuild }}
```

Comment:
Do you need to include this template again here if it's already defined above / included in this file itself??

Reply:
Yes, when including another template you need to pass any parameters explicitly. The fact that this file takes a parameter with the same name doesn't matter when including another one.